scapy/route6.py

```diff
@@ -200,7 +201,7 @@ def route(self, dst, dev=None):
         # Deal with dev-specific request for cache search
         k = dst
         if dev is not None:
-            k = dst + "%%" + dev
+            k = dst + "%%" + (dev if isinstance(dev, basestring) else dev.pcap_name)
```

Comment:
It's dev.pcap_name right ? Cause i wasn't sure if it shouldn't have been dev.name...

> p-l-: Cannot remember... @guedou?